pkg/runtime/wfengine/activity.go

```diff
@@ -16,26 +16,55 @@ package wfengine
 
 import (
 	"context"
+	"encoding/gob"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"strings"
+	"sync"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/microsoft/durabletask-go/api"
 	"github.com/microsoft/durabletask-go/backend"
 
 	"github.com/dapr/dapr/pkg/actors"
 	invokev1 "github.com/dapr/dapr/pkg/messaging/v1"
 )
 
+var ErrDuplicateInvocation = errors.New("duplicate invocation")
+
 type activityActor struct {
-	actorRuntime actors.Actors
-	scheduler    workflowScheduler
+	actorRuntime     actors.Actors
+	scheduler        workflowScheduler
+	statesCache      sync.Map
+	cachingDisabled  bool
+	defaultTimeout   time.Duration
+	reminderInterval time.Duration
+}
+
+// ActivityRequest represents a request by a worklow to invoke an activity.
+type ActivityRequest struct {
+	HistoryEvent []byte
+	Generation   uuid.UUID
```

Comment:
Should these be somehow related to one another like monotonically increasing numbers or timestamps? Would I ever need to know that one is older than another -- maybe for the purposes of autocancelling an older generation if I have seen a newer generation, or viewing them in time-order as a human?

Reply:
Yeah, that might actually be useful to simplify debugging. I chose UUID because the Durable Task Framework uses UUIDs for this same concept, but I don't think it's strictly necessary.